### PR TITLE
Check image is already present before pull from library

### LINKED ISSUE #257: Query library image hash before pulling
So that pull policy always will work in an optimal way

### PR DESCRIPTION
**Pull request by sashayakovtseva**
Closes #257

### CODE REVIEW

File: pkg/image/image.go [hunk 1]
```diff
@@ -186,6 +190,40 @@ func Pull(location string, ref *Reference, auth *k8s.AuthConfig) (img *Info, err
 	}, nil
 }
 
+// LibraryInfo queries remote library to get info about the image.
+// If image is not found returns ErrNotFound. For references other than
+// library returns nil, nil.
```
Comment (cali4888) [inline pkg/image/image.go]: but if user sends a references other than library it have to be an error, isn't it?

Comment (cali4888) [reply pkg/image/image.go]: returning nil, nil it's not a really good practice

File: pkg/image/image.go [hunk 2]
```diff
@@ -186,6 +190,40 @@ func Pull(location string, ref *Reference, auth *k8s.AuthConfig) (img *Info, err
...
+	}
+	img, found, err := client.GetImage(ctx, pullURL)
+	if err != nil {
+		return nil, fmt.Errorf("could not get library image info: %v", err)
+	}
+	if !found {
+		return nil, ErrNotFound
+	}
+
+	id := strings.TrimPrefix(img.Hash, "sha256.")
```
Comment (cali4888) [inline pkg/image/image.go]: what it the purpose why you trimming `sha256.` maybe comment here?

File: pkg/image/image.go (hunk 1 above)
Comment (cali4888) [reply pkg/image/image.go]: but why you returning nil? users provides an invalid input to the function. It's an error

Comment (cali4888) [reply pkg/image/image.go]: I would better return ErrNotLibraryDomain or similar

Comment (sashayakovtseva) [reply pkg/image/image.go]: input is an arbitrary reference and docker reference is still valid
the main idea in returning nil,nil is to say "hey, no library image corresponds to your reference"

**Pull request merged**